control/init.sh

```diff
@@ -17,14 +17,24 @@ function realpath() {
 
 
 if [ -z ${CI_TEST} ]; then
-    if [ ! -f $VENV/bin/activate ]; then
-        if [[ $CCHQ_VIRTUALENV == *3.10* ]]; then
-          # use venv because 3.10 setup includes installing python3.10-venv
-          python3.10 -m venv $VENV
+    if [[ $(cat /etc/os-release | grep '^VERSION=' | tr -dc '0-9.') == *18.04* ]] && hash python3.10 2>/dev/null; then
```

Comment:
I think `VERSION_ID` is the one you want. The fact that this file is meant to be a shell-sourceable makes grepping it annoying (you have to contend with quote characters that may or may not be present). Perhaps just source it in a subshell?

```suggestion
    if hash python3.10 2>/dev/null && [[ $( source /etc/os-release; echo $VERSION_ID ) == 18.04 ]]; then
```

Reply:
See 38a57d19c00901b8085204834bff68134cf46e0f